src/kakarot/instructions/environmental_information.cairo

```diff
@@ -564,6 +564,18 @@ namespace EnvironmentalInformation {
         tempvar address = new model.Address(starknet_address, evm_address);
         let (state, account) = State.get_account(ctx.state, address);
 
+        // Relevant cases:
+        // https://github.com/ethereum/go-ethereum/blob/master/core/vm/instructions.go#L392
+        let (state, balance) = State.read_balance(state, address);
+        let has_code_or_nonce = Account.has_code_or_nonce(account);
```

Comment:
I'd invert the order of the lines plz, it's a bit harder to follow like this, because rn

1. we get the account
2. we get the balance
3. we use the account to get has_code_or_nonce

doing

1. we get the account
2. we use the account to get has_code_or_nonce
3. we get the balance
4. we use the balance and has_code_or_nonce

looks smoother